src/ldp/representation/RepresentationMetadata.ts

```diff
@@ -1,47 +1,128 @@
-/**
- * Contains metadata relevant to a representation.
- */
-import { Quad } from 'rdf-js';
+import { Store } from 'n3';
+import { BlankNode, Literal, NamedNode, Quad, Term } from 'rdf-js';
+import { quad as createQuad, literal, namedNode } from '@rdfjs/data-model';
 
 /**
- * Metadata corresponding to a {@link Representation}.
+ * Stores the metadata triples and provides methods for easy access.
  */
-export interface RepresentationMetadata {
+export class RepresentationMetadata {
+  private store: Store;
+  private id: NamedNode | BlankNode;
+
   /**
-   * All metadata triples of the resource.
+   * @param identifier - Identifier of the resource relevant to this metadata.
+   *                     A blank node will be generated if none is provided.
+   *                     Strings will be converted to named nodes.
+   * @param quads - Quads to fill the metadata with.
    */
-  raw: Quad[];
+  public constructor(identifier?: NamedNode | BlankNode | string, quads?: Quad[]) {
+    this.store = new Store(quads);
+    if (identifier) {
+      if (typeof identifier === 'string') {
+        this.id = namedNode(identifier);
+      } else {
+        this.id = identifier;
+      }
+    } else {
+      this.id = this.store.createBlankNode();
```

Comment:
Do we really want to accept bnodes here?
In what cases can those occur?

Reply:
A specific case for that is [here](https://github.com/solid/community-server/pull/65/files#diff-d076801681c3d3350fa0c05b6278fea4R22), where we want to create new metadata based on input metadata (because we don't want to change the original). In this case we don't know if the identifier is a bnode or not when we do this.

Reply:
Ok, but that's because this also refers to `id` of `RepresentationMetadata`.
So if we fix that to `NamedNode`, it should be solved, right?

Or is there another place where the id can be set to a bnode?

Reply:
Ah the comment was about the id in general. A bnode is used when the identifier of the resource it not yet known when the metadata is made. This is for example the case in the BodyParser since that one just makes the representation while another component is making the identifier. Another case is for a POST where you don't know what the final identifier is going to look like.

Another option would be to use a fixed named node when the identifier is unknown.

Reply:
This makes sense if we really need to represent this metadata internally as triples (which I'm still not 100% convinced of).
But it feels like a hacky way to get it in here (and potentially unsafe). If a hash-based representation would be used, the metadata entries would be stored in a hash, independently of the id, which would only be set once it's known.

Reply:
Just realized something that I didn't mention in my other comment below. But I think it should be possible to have metadata triples where the subject is not the identifier (depending on the ontology probably, and perhaps if we want to store more complex headers which have several properties). And that is also done easier by having an internal store.

Reply:
Ok, that's a strong argument in favor of triples indeed :-)

However, we could have a `MetadataSerializer` that converts from a hash-based representation to RDF (and could have different implementations, e.g. different ontologies). But may be overkill, so still no strong opinion, just open questions.

Reply:
Isn't it the other way around? Since we store metadata as triples we are in fact dependent on the used ontology.

Reply:
I meant when having to add functionality to the metadata to support a new ontology. True, if we change the ontology, the relevant classes would have to be updated, but imo that is less of an issue.